plugins/CustomVariables/config/config.php

```diff
@@ -1,9 +1,27 @@
 <?php
 
+use Piwik\Plugins\CustomVariables\Model;
+
 return array(
 
     'tracker.request.processors' => DI\add(array(
         DI\get('Piwik\Plugins\CustomVariables\Tracker\CustomVariablesRequestProcessor'),
     )),
 
+    'Piwik\Plugins\CustomVariables\Model.page' => DI\object('Piwik\Plugins\CustomVariables\Model')
+        ->constructor(Model::SCOPE_PAGE),
+    'Piwik\Plugins\CustomVariables\Model.visit' => DI\object('Piwik\Plugins\CustomVariables\Model')
+        ->constructor(Model::SCOPE_VISIT),
+    'Piwik\Plugins\CustomVariables\Model.conversion' => DI\object('Piwik\Plugins\CustomVariables\Model')
+        ->constructor(Model::SCOPE_CONVERSION),
+
+    'Piwik\Plugins\CustomVariables\Model.all' => array(
```

Comment:
It's not about changing/extending the code (not always), it's about decoupling dependencies. Model is a dependency of CustomVariableMetadataProvider, so Model has to be in DI to inject it. If we don't do this everywhere there's application logic, we'll have to keep using StaticContainer in classes that aren't in DI, which will make embedding environments (important for cloud & tests) impossible to do completely.

The reason for the complexity here is due to the design of the Model class. Since it takes a scope as a constructor argument, we have to create multiple instances of it in DI. I can simplify it w/ some refactoring, just didn't want to w/ this PR. Of course, since the PR is broken, it's not as big a deal now.